### PR TITLE
Some initial cleanups of the config code to prepare for the larger refactoring.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Extracted from #141.

### CODE REVIEW

File: confgenerator/fluentbit/conf_test.go [hunk 1]
```diff
@@ -296,9 +296,9 @@ func TestTail(t *testing.T) {
 			expectedTailConfig: `[INPUT]
```
Comment (davidbtucker) [inline confgenerator/fluentbit/conf_test.go:296]: Btw, I would like to get rid of all these unit tests and just use the file-based tests.

File: confgenerator/config.go [hunk 2]
```diff
@@ -395,9 +385,14 @@ func collectYamlFields(s interface{}) []yamlField {
 				// Expand inline structs.
 				parameters = append(parameters, recurse(v)...)
 			} else if f.PkgPath == "" { // skip private non-struct fields
+				t, e := f.Tag.Lookup("validate")
```
Comment (davidbtucker) [inline confgenerator/config.go:388]: This could just be:

    t, _ := ...

**Review by davidbtucker — APPROVED** [review]

File: confgenerator/config.go (hunk 2 above)
Comment (igorpeshansky) [reply confgenerator/config.go]: Done.

File: confgenerator/fluentbit/conf_test.go (hunk 1 above)
Comment (igorpeshansky) [reply confgenerator/fluentbit/conf_test.go:296]: Yep, future PRs.

**Pull request merged**